plugins/modules/device_credential_intent.py

```diff
@@ -0,0 +1,2323 @@
+                    params=credential_params,
+                )
+                self.log(str(response))
+                if not response:
+                    self.msg = "response is empty"
+                    self.status = "exited"
+                    return self
+
+                if not isinstance(response, dict):
+                    self.msg = "response is not a dictionary"
+                    self.status = "exited"
+                    return self
+
+                task_id = response.get("response").get("taskId")
+                while True:
+                    task_details = self.get_task_details(task_id)
+                    self.log(str(task_details))
+                    if task_details.get("isError") is True:
+                        self.msg = str(task_details.get("progress"))
+                        self.status = "failed"
+                        return self
+                    elif task_details.get("isError") is False:
+                        self.result['changed'] = True
+                        break
+        self.log("Update Device Credential API input - " + str(final_response))
+        self.log("Global Device Credential Updated Successfully")
+        result_global_credential.update({
+            "Updation": {
+                "response": final_response,
+                "msg": "Global Device Credential Updated Successfully"
+            }
+        })
+        self.msg = "Global Device Credential Updated Successfully"
+        self.status = "success"
+        return self
+
+    def assign_credentials_to_site(self):
+        """
+        Assign Global Device Credential to the Cisco DNA
+        Center based on the provided playbook details.
+        Check the return value of the API with check_return_status().
+
+        Parameters:
+            None
+
+        Returns:
+            self
+        """
+
+        result_assign_credential = self.result.get("response")[0].get("assignCredential")
+        credential_params = self.want.get("assign_credentials")
+        final_response = []
+        self.log("Assign Device Credential to site API input - " + str(credential_params))
+        if not credential_params:
+            result_assign_credential.update({
+                "No Assign Credentials": {
+                    "response": "No Response",
+                    "msg": "No Assignment is available"
+                }
+            })
+            self.msg = "No Assignment is available"
+            self.status = "success"
+            return self
+        for site_id in self.want.get("site_id"):
+            credential_params.update({"site_id": site_id})
+            final_response.append(credential_params)
+            response = self.dnac._exec(
+                family="network_settings",
+                function='assign_device_credential_to_site_v2',
+                params=credential_params,
+            )
+            self.log(str(response))
+            if not response:
```

Comment:
We can also use base class for it.. I see that this code is repeated
```
def validate_api_response(self, response):
    if not response:
        self.msg = "response is empty"
        self.status = "failed"
        return False

    if not isinstance(response, dict):
        self.msg = "response is not a dictionary"
        self.status = "failed"
        return False

    return True
```
And we can simply replace this in all our modules by simply calling
```
     if not self.validate_api_response(response):
            return self
```